NeXus/NXptycho.nxdl.xml

```diff
@@ -0,0 +1,180 @@
+  # License as published by the Free Software Foundation; either
+  # version 3 of the License, or (at your option) any later version.
+  #
+  # This library is distributed in the hope that it will be useful,
+  # but WITHOUT ANY WARRANTY; without even the implied warranty of
+  # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  # Lesser General Public License for more details.
+  #
+  # You should have received a copy of the GNU Lesser General Public
+  # License along with this library; if not, write to the Free Software
+  # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+  #
+  # For further information, see http://www.nexusformat.org
+-->
+<definition name="NXptycho" extends="NXobject" type="group"
+			category="application"
+			xmlns="http://definition.nexusformat.org/nxdl/3.1"
+			xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    		xsi:schemaLocation="http://definition.nexusformat.org/nxdl/3.1 ../nxdl.xsd"
+    >
+    <symbols>
+        <doc>
+            These symbols will be used below to coordinate the shapes of the datasets.
+        </doc>
+    </symbols>
+
+    <doc>
+		Application definition for ptychography experiments.
+    </doc>
+
+    <group type="NXentry" name="entry_1">
+    	<field name="definition" type="NX_CHAR" minOccurs="1" maxOccurs="1">
+    		<doc>
+                Official NeXus NXDL schema to which this file conforms
+            </doc>
+    		<enumeration>
+    			<item value="NXptycho"/>
+    		</enumeration>
+		</field>
+        <field name="experiment_description" type="NX_CHAR" minOccurs="0" maxOccurs="1">
+            <doc>
+                Brief summary of the experiment, including key objectives. String to define which ptychography
+                type was performed, e.g. basic, bragg, fourier, ...
+            </doc>
+        </field>
+		<field name="entry_identifier">
+			<doc>
+				Unique identifier for the experiment.
+			</doc>
+		</field>
+    	<field name="entry_identifier_uuid" minOccurs="0" maxOccurs="1">
+			<doc>
+        		(optional) UUID identifier for this entry.
+			</doc>
+    	</field>
+		<field name="entry_number" type="NX_INT">
+			<doc>
+				Entry or scan number (must be an integer).
+				NOTE: Link to collection_identifier.
+			</doc>
+		</field>
+
+    	<group type="NXinstrument" name="instrument_1" minOccurs="1" maxOccurs="1">
+    		<group type="NXbeam" name="beam_1" minOccurs="1" maxOccurs="1">
+    			<field name="incident_beam_energy" type="NX_FLOAT" minOccurs="1" maxOccurs="1">
+					<doc>Photon energy of the beam on entering the sample</doc>
+    				<attribute name="units" type="NX_CHAR" optional="false"/>
+    			</field>
+                <field name="incident_beam_wavelength" type="NX_FLOAT" minOccurs="1" maxOccurs="1">
+					<doc>Wavelength of the beam on entering the sample</doc>
+    				<attribute name="units" type="NX_CHAR" optional="false"/>
+    			</field>
```

Comment:
Both beam energy and wavelength are listed as required. Only one should be required?

Reply:
done